### PR TITLE
Fix DataTableV3 serde bug for empty array

### PR DESCRIPTION
**Pull request by xiangfu0**
Caught this exception in [DataTableV4 fix](https://github.com/apache/pinot/pull/10543/files#diff-2276298496d8c930553150ded2ebe21d7caa2b809e450a98618d1bd655c04891R153) , apparently, it happens in V3 as well.

### CODE REVIEW

**Comment by xiangfu0** [comment]
@mqliang

File: pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java [hunk 1]
```diff
@@ -123,36 +123,142 @@ public void testException(int dataTableVersion)
...
   @Test(dataProvider = "versionProvider")
-  public void testEmptyStrings(int dataTableVersion)
+  public void testEmptyValues(int dataTableVersion)
       throws IOException {
     DataTableBuilderFactory.setDataTableVersion(dataTableVersion);
     String emptyString = StringUtils.EMPTY;
     String[] emptyStringArray = {StringUtils.EMPTY};
+    ByteArray emptyBytes = new ByteArray(new byte[0]);
+    for (int numRows = 0; numRows < NUM_ROWS; numRows++) {
+      testEmptyValues(new DataSchema(new String[]{"STR_SV", "STR_MV"}, new DataSchema.ColumnDataType[]{
```
Comment (walterddr) [inline pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java:133]: i dont know if this is possible but lets also test empty `OBJECT` type?

File: pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java [hunk 2]
```diff
@@ -123,36 +123,142 @@ public void testException(int dataTableVersion)
...
+      testEmptyValues(
+          new DataSchema(new String[]{"BYTES"}, new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.BYTES}),
+          numRows, new Object[]{emptyBytes});
+
+      testEmptyValues(
+          new DataSchema(new String[]{"BOOL_ARR"},
+              new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.BOOLEAN_ARRAY}),
+          numRows, new Object[]{new int[]{}});
+
+      testEmptyValues(
```
Comment (walterddr) [inline pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java:154]: why is this one initialized with a `{1}`?

File: pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java:133]: We don't really use `OBJECT` type anywhere and what is an empty `OBJECT` if not null?

File: pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java (hunk 2 above)
Comment (xiangfu0) [reply pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java:154]: Wanna test array with 0 element and 1 element with default value.

Comment (xiangfu0) [reply pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java:154]: this should be `new int[]{0}` for boolean default

**Review by walterddr — APPROVED** [review]

**Review by Jackie-Jiang — COMMENTED** [review]
Seems we handle `null` value everywhere, and the only place not handled is `BaseDataTable` constructor. Should we fix that instead?

**Comment by xiangfu0** [comment]
> Seems we handle `null` value everywhere, and the only place not handled is `BaseDataTable` constructor. Should we fix that instead?

I will merge this one for now. Will let @61yao  to decide if the `ByteBuffer` based constructor should be moved to `BaseDataTable`.

**Pull request merged**